modules/claims_api/app/controllers/claims_api/v2/veteran_identifier_controller.rb

```diff
@@ -5,18 +5,19 @@
 module ClaimsApi
   module V2
     class VeteranIdentifierController < ClaimsApi::V2::ApplicationController
-      # TODO: REMOVE BEFORE IMPLEMENTATION
-      skip_before_action :authenticate, only: %i[find]
-      ICN_FOR_TEST_USER = '1012667145V762142'
-      # TODO: REMOVE BEFORE IMPLEMENTATION
-
       def find
         raise ::Common::Exceptions::Unauthorized if request.headers['Authorization'].blank?
 
         validate_request
         veteran = find_veteran(params)
 
-        render json: { id: veteran[:id] }
+        raise ::Common::Exceptions::ResourceNotFound unless veteran_icn_found?(veteran)
+
+        user_is_the_veteran   = current_user_is_the_veteran?(veteran: veteran, user: @current_user)
+        user_is_a_veteran_rep = current_user_is_a_veteran_representative?(@current_user)
+        raise ::Common::Exceptions::Unauthorized unless user_is_the_veteran || user_is_a_veteran_rep
```

Comment:
This should be a 403, maybe the ::Common::Exceptions::Forbidden class?
Because the user has authenticated fine, but they are forbidden from getting this Veteran's ID

Reply:
Ah, that's a really good point!  
Good catch.